src/parser/jobs/ast/modules/ArcanaTracking/ArcanaTracking.tsx

```diff
@@ -198,59 +177,55 @@ export default class ArcanaTracking extends Module {
 	/**
 	 * Adds the Arcana Buff to _prepullArcanas if it was a precast status
 	 */
-	private onPrepullArcana(event: BuffEvent) {
-		if (event.timestamp > this.parser.fight.start_time) {
+	private onPrepullArcana(event: Events['statusApply']) {
+		if (event.timestamp > this.parser.pull.timestamp) {
 			return
 		}
-		this.prepullArcanas.push(event)
+		if (this.on_prepullArcanas != null) {
+			this.on_prepullArcanas.status.push(event)
+		}
```

Comment:
Splitting the prepullArcanas array into separate apply/remove arrays breaks this functionality, since we never set anything into this.off_prepullArcanas and thus the offPrepullArcana event now will never do anything.  Instead of changing to the interfaces and the split arrays, you should change `this.prepullArcanas` to be an `Array<Events['statusApply']>` and keep most of the previous behavior (with the timestamp handling adjusted accordingly)

Reply:
I think I updated this appropriately based on feedback
please let me know if this wasn't the intended update